packages/admin/blocks-admin/src/common/useScrollRestoration.ts

```diff
@@ -20,6 +20,14 @@ export function useScrollRestoration<Element extends HTMLElement>(identifier: st
         }
     }, [identifierForRoute]);
 
+    for (const key of Object.keys(scrollPositions)) {
+        const keyWithoutIdentifier = key.replace(`-${identifier}`, "");
+
+        if (keyWithoutIdentifier.startsWith(location.pathname) && keyWithoutIdentifier !== location.pathname) {
+            scrollPositions[key] = 0;
+        }
```

Comment:
I believe this is too complicated, shouldn't

```ts
if (!key.startsWith(location.pathname)) {
  // Reset position
  delete scrollPositions[key];
}
```

work as well?

Also, should this be done on every render? How often is this run?

Reply:
No, that does not work. You want to delete the scrollPosition when you navigate out from the nested route. That is the case if the pathname of the nested route starts with the pathname of the other route. You also need to check that it is not the same so it does not delete the scroll position of itself.

First route: "route-x/".
Nested route: "route-x/nested"

You also need to remove the identifier cause for the comparison check since it would be:

First route with identifier: "route-x/identifier"
Nested route with identifier: "route-x/nested/identifier"

I found just removing the identifier to be the easiest solution to enable comparing the paths.

-------

As for how often it runs, it only runs on path changes: 

https://github.com/vivid-planet/comet/assets/118435139/a7048474-f678-4ba7-b299-2b14a6ba2de1

![Screenshot 2024-02-06 at 12 04 22](https://github.com/vivid-planet/comet/assets/118435139/e336f280-bbed-436e-81be-b760360c2d7a)

Reply:
Resolved: outdated code.